src/detect-engine-build.c

```diff
@@ -1425,7 +1425,13 @@ static inline int CreatePortList(DetectEngineCtx *de_ctx, const uint8_t *unique_
                 port = port2 + 1;
             } else if (p1 && p1->single) {
                 SCPortIntervalFindOverlappingRanges(de_ctx, port, port, &it->tree, list);
```

Comment:
Could this be done only in the `else` case ?

Reply:
I looked the code up and tried. But, the order is important here, we first want to check if the first port has certain properties like if it is single, so, I think it cannot be done in the else..